test-cases/gutenberg/group.md

```diff
@@ -0,0 +1,228 @@
+-   Nest some blocks inside
+-   Check if margins and paddings do not bounce when changing selection (exception is the trasition of the selected block connected with making space for FloatingToolbar)
+-   Check if there is no weird/unexpected extra free space around blocks
+-   Check the margins/paddings values
+
+Expected look:  
+![MarginRoot](../resources/group-margin-root.png)
+ - Margins of `3px` left/right (between screen edge and solid border)
+
+![MarginGroupChild](../resources/group-margin-inner-child-group.png)
+ - Margins of `5px` left/right (beetwen dashed border and solid border)
+
+![MarginChild](../resources/group-margin-child.png)
+ - Margin of `6px` left/right (beetwen content and dashed border)
+
+![MarginContent](../resources/group-margin-root-1.png)
+ - Margin of `16px` left/right
+
+Total space from screen edge to content should equals `16px` in each state (including solid and dashed border width in selection state)
+
+--------------------------------------------------------------------------------
+
+##### TC008
+
+### Nested selection cause applying dimmed style on the rest of blocks
+
+-   Add a `Group` block
+-   Create some nested structure
+-   Navigate down in the hierarchy
+-   Check if each time you change selection all blocks that are higher in hierarchy tree gets dimmed
+-   Select `Group` block
+-   Check if none block gets dimmed
+
+Expected look:  
+![Dimming1](../resources/group-dimming-1.png)
+  
+![Dimming2](../resources/group-dimming-2.png)
+  
+![Dimming3](../resources/group-dimming-3.png)
+
+--------------------------------------------------------------------------------
+
+##### TC009
+
+### Breadcrumbs on FloatingToolbar is properly displayed
+
+-   Add a `Group` block
+-   Create nested structure
+-   Check if each time you change selection `Breadcrumbs` show focused block title and focused block icon
+-   Check if each time you change selection `Breadcrumbs` show top most parent block icon
+-   Check if `Breadcrumbs` renders navigation up button
+
+Expected look:  
+![BreadcrumbsRoot](../resources/group-breadcrumbs-root.png)
+  
+![BreadcrumbsNested](../resources/group-breadcrumbs-nested.png)
+
+--------------------------------------------------------------------------------
+
+##### TC010
+
+### Navigation up button works as expected
+
+-   Add a `Group` block
+-   Create nested structure
+-   Select deep nested block
+-   Press navigation up button on `Breadcrumb`
+-   Check if each time you press navigation up button the selection moves to parent block
+-   Check if border and dim styles changes accordingly
+
+--------------------------------------------------------------------------------
+
+##### TC011
+
+### Navigation down works as expected
+
+-   Add a `Group` block
+-   Create some nested structure ( at least 3 levels deep )
+-   Clear selection
+-   Press on the bottom-most block in hierarchy of added `Group`
+-   Check if the selection is redirect properly to top-most parent
```

Comment:
wdym by the `bottom-most block in hierarchy `? 
I am not following how next step of pressing the bottom most block is redirecting us to `top-most parent `

Reply:
If you have the Group Unselected and press on Block which is on very deep level in hierarchy (in above I name it bottom-most in hierarchy). The selection always be redirected to top-most parent of this block. In this case the Group block itself. Top-most in above means the parent on the top of hierarchy tree

If you repeat the same press. The selection goes to top-most parent of same block which is also a child of selected block. In other words the logic there finds the CommonAncestor between pressed block and selected block

Reply:
ok so basically we are trying to emphasize the parent first approach